source/python/convert_intrinsics_opencv_to_halcon.py

```diff
@@ -0,0 +1,297 @@
+                f"Sy:sy:   {self.Sx:.06e};",
+                f"  DOUBLE:0.0:;",
+                f'  "Height of a cell on the sensor";',
+                f"",
+                f"Cx:cx:   {self.Cx:.06e};",
+                f"  DOUBLE::;",
+                f'  "X-coordinate of the image center";',
+                f"",
+                f"Cy:cy:   {self.Cy:.06e};",
+                f"  DOUBLE::;",
+                f'  "Y-coordinate of the image center";',
+                f"",
+                f"ImageWidth:imgw:  {self.ImageSize[0]};",
+                f"  INT:1:32768;",
+                f'  "Width of the images";',
+                f"",
+                f"ImageHeight:imgh:  {self.ImageSize[1]};",
+                f"  INT:1:32768;",
+                f'  "Height of the images";',
+            ]
+        )
+
+    def _write_to_dat_file(self, filepath: Path) -> None:
+        if not exists(filepath):
+            with open(filepath, "w") as text_file:
+                text_file.write(self._get_dat_text())
+        else:
+            raise Exception(f"{filepath} already exists!")
+
+
+@dataclass
+class ZividCameraParameters:
+    pixel_size: NDArray[float]
+    image_size: NDArray[int]
+
+    @staticmethod
+    def zivid_one() -> ZividCameraParameters:
+        return ZividCameraParameters(pixel_size=5.86e-6, image_size=[1920, 1200])
+
+    @staticmethod
+    def zivid_one_plus() -> ZividCameraParameters:
+        return ZividCameraParameters.zivid_one()
+
+    @staticmethod
+    def zivid_two() -> ZividCameraParameters:
+        return ZividCameraParameters(pixel_size=4.50e-6, image_size=[1944, 1200])
+
+    @staticmethod
+    def from_name(name: str) -> ZividCameraParameters:
+        if name.lower() == "zivid one":
+            return ZividCameraParameters.zivid_one()
+        if name.lower() == "zivid one plus":
+            return ZividCameraParameters.zivid_one_plus()
+        if name.lower() == "zivid one plus s":
+            return ZividCameraParameters.zivid_one_plus()
+        if name.lower() == "zivid one plus m":
+            return ZividCameraParameters.zivid_one_plus()
+        if name.lower() == "zivid one plus l":
+            return ZividCameraParameters.zivid_one_plus()
+        if name.lower() == "zivid two":
+            return ZividCameraParameters.zivid_two()
+        raise Exception(f"{name} not a valid model name")
+
+
+def _cost_function(
+    poly: NDArray[5],
+    halcon_parameters: HalconParameters,
+    opencv_parameters: OpenCVParameters,
+    pixels: NDArray[Any, Any, 2],
+) -> float:
+    halcon_parameters.Poly = poly
+    halcon_pixels_undistorted = halcon_parameters.undistorted_pixels(pixels=pixels)
+    opencv_pixels_undistorted = opencv_parameters.undistorted_pixels(pixels=pixels)
+    return np.sqrt(np.mean(np.linalg.norm(halcon_pixels_undistorted - opencv_pixels_undistorted, axis=2) ** 2))
+
+
+def _create_pixels(zivid_camera_parameters: ZividCameraParameters, samples_x: float = 25) -> NDArray[Any, 1, 2]:
```

Comment:
same comment here about the optional argument.

Reply:
done